src/TypeInference/Expr.ml

```diff
@@ -151,19 +151,20 @@ and method_call_ctx ~pos ~env ~self ~self_tp ~self_r_eff ~eff =
   | TArrow(
       { sch_targs = []; sch_named = []; sch_body = self_tp' },
       res_tp, res_eff) ->
-    if not (Unification.subtype env self_tp self_tp') then
-      Error.report
-        (Error.expr_type_mismatch ~pos:self.pos ~env self_tp self_tp');
-    if not (Unification.subeffect env res_eff eff) then
-      Error.report (Error.method_effect_mismatch ~pos ~env res_eff eff);
+    let unification_result = Unification.subtype env self_tp self_tp'
+      in Error.check_unify_result ~is_fatal:false ~pos unification_result
+        ~on_error:(Error.expr_type_mismatch ~pos:self.pos ~env self_tp self_tp');
+    let unification_result = Unification.subeffect env res_eff eff
+      in Error.check_unify_result ~is_fatal:false ~pos unification_result
+        ~on_error:(Error.method_effect_mismatch ~pos:pos ~env res_eff eff);
```

Comment:
Code formatting is inconsistent with existing code. This comment also applies to other changes in this file and `Pattern.ml`